src/plugins/embeddable/public/add_from_library/registry.ts

```diff
@@ -12,50 +12,43 @@ import { i18n } from '@kbn/i18n';
 import { CanAddNewPanel } from '@kbn/presentation-containers';
 import { FinderAttributes, SavedObjectCommon } from '@kbn/saved-objects-finder-plugin/common';
 import { SavedObjectMetaData } from '@kbn/saved-objects-finder-plugin/public';
+import { useMemo } from 'react';
 
-type SOToEmbeddable<TSavedObjectAttributes extends FinderAttributes = FinderAttributes> = (
-  container: CanAddNewPanel,
-  savedObject: SavedObjectCommon<TSavedObjectAttributes>
-) => void;
-
-export type ReactEmbeddableSavedObject<
-  TSavedObjectAttributes extends FinderAttributes = FinderAttributes
-> = {
-  onAdd: SOToEmbeddable<TSavedObjectAttributes>;
+export type RegistryItem<TSavedObjectAttributes extends FinderAttributes = FinderAttributes> = {
+  onAdd: (
+    container: CanAddNewPanel,
+    savedObject: SavedObjectCommon<TSavedObjectAttributes>
+  ) => void;
   savedObjectMetaData: SavedObjectMetaData;
 };
 
-const registry: Map<string, ReactEmbeddableSavedObject<any>> = new Map();
+const registry: Map<string, RegistryItem<any>> = new Map();
 
-export const registerReactEmbeddableSavedObject = <
-  TSavedObjectAttributes extends FinderAttributes
->({
+export const registerAddFromLibraryType = <TSavedObjectAttributes extends FinderAttributes>({
   onAdd,
-  embeddableType,
   savedObjectType,
   savedObjectName,
   getIconForSavedObject,
   getSavedObjectSubType,
   getTooltipForSavedObject,
 }: {
-  onAdd: SOToEmbeddable<TSavedObjectAttributes>;
-  embeddableType: string;
+  onAdd: RegistryItem['onAdd'];
   savedObjectType: string;
   savedObjectName: string;
   getIconForSavedObject: (savedObject: SavedObjectCommon<TSavedObjectAttributes>) => IconType;
   getSavedObjectSubType?: (savedObject: SavedObjectCommon<TSavedObjectAttributes>) => string;
   getTooltipForSavedObject?: (savedObject: SavedObjectCommon<TSavedObjectAttributes>) => string;
 }) => {
-  if (registry.has(embeddableType)) {
+  if (registry.has(savedObjectType)) {
     throw new Error(
       i18n.translate('embeddableApi.embeddableSavedObjectRegistry.keyAlreadyExistsError', {
         defaultMessage: `Embeddable type {embeddableType} already exists in the registry.`,
```

Comment:
nit. Should we add more specificity to the registry that is throwing the error?

```suggestion
        defaultMessage: `Saved object type {savedObjectType} already exists in the Add from Library registry.`,
```

Reply:
Resolved with 5adbf55978aa082c36ba9e83a9da8eecc12c6bd1. I also remove i18n token since this error message is a developer facing string and not a UI facing string.